qemu/tests/block_resize.py

```diff
@@ -87,85 +90,67 @@ def compare_block_size(session, block_cmd, block_pattern):
     block_size_cmd = params["block_size_cmd"].format(drive_path)
     block_size_pattern = params.get("block_size_pattern")
     need_reboot = params.get("need_reboot", "no") == "yes"
+    need_rescan = params.get("need_rescan", "no") == "yes"
     accept_ratio = float(params.get("accept_ratio", 0))
-
-    error_context.context("Check image size in guest", logging.info)
-    block_size = get_block_size(session, block_size_cmd, block_size_pattern)
-    if (block_size > data_image_size or
-            block_size < data_image_size * (1 - accept_ratio)):
-        test.error("Image size from guest and image not match"
-                   "Block size get from guest: %s \n"
-                   "Image size get from image: %s \n"
-                   % (block_size, data_image_size))
+    drive_letter = params.get("drive_letter", "C")
 
     if params.get("guest_prepare_cmd"):
         session.cmd(params.get("guest_prepare_cmd"))
 
     if params.get("format_disk", "no") == "yes":
         error_context.context("Format disk", logging.info)
-        utils_misc.format_windows_disk(session, params["disk_index"],
-                                       mountpoint=params["disk_letter"])
-
-    disk_update_cmd = params.get("disk_update_cmd")
-    if disk_update_cmd:
-        disk_update_cmd = disk_update_cmd.split("::")
-
-    disk_rescan_cmd = params.get("disk_rescan_cmd")
-
-    block_size = data_image_size
-    disk_change_ratio = params["disk_change_ratio"]
-    for index, ratio in enumerate(disk_change_ratio.strip().split()):
-        old_block_size = block_size
-        block_size = int(int(data_image_size) * float(ratio))
-
-        if block_size == old_block_size:
-            logging.warn("Block size is not changed in round %s."
-                         " Just skip it" % index)
-            continue
-
-        if disk_update_cmd:
-            if "DISK_CHANGE_SIZE" in disk_update_cmd[index]:
-                disk_unit = params.get("disk_unit", "M")
-                size = abs(block_size - old_block_size)
-                change_size = utils_misc.normalize_data_size("%sB" % size,
-                                                             disk_unit)
-                disk_update_cmd[index] = re.sub("DISK_CHANGE_SIZE",
-                                                change_size.split(".")[0],
-                                                disk_update_cmd[index])
-
-        # So far only virtio drivers support online auto block size change in
-        # linux guest. So we need manully update the the disk or even reboot
-        # guest to get the right block size after change it from monitor.
+        drive_letter = utils_disk.configure_empty_windows_disk(session,
```

Comment:
if windows disk is empty, need to clear readonly attribute and online it. refer to #1607